Groups/GroupFinder.ascx.cs

```diff
@@ -1346,10 +1344,9 @@ private void SetGroupTypeOptions()
 
         private void AddToFilterOrder( int defaultkey, string filterId )
         {
-            var filterAddKey = FilterOrder.FirstOrDefault( f => f.Value == filterId ).Key;
-            if ( !FilterOrder.ContainsKey( ( filterAddKey != 0 ) ? filterAddKey : defaultkey ) )
+            if ( !FilterOrder.ContainsValue( filterId ) )
```

Comment:
This needs more thought, this will not accommodate when an attribute gets deleted.

Reply:
Updated